components/eventing-controller/controllers/subscription/nats/reconciler_test.go

```diff
@@ -71,76 +70,93 @@ var (
 	}
 )
 
+// testNATSUnavailabilityReflectedInSubscriptionStatus tests if the reconciler can correctly resolve a Subscription in
+// the case of a NATS server that becomes unavailable.
+// The test is conducted in the following steps:
+// 1. the NATS server that is available
+// 2. the NATS server that is no longer available
+// 3. the NATS server is available again
 func testNATSUnavailabilityReflectedInSubscriptionStatus(id int, eventTypePrefix, _, eventTypeToSubscribe string) bool {
 	return When("NATS server is not reachable and max retries are exceeded", func() {
-		It("Should mark the Subscription as not ready until NATS is reachable again", func() {
-			subscriptionName := fmt.Sprintf(subscriptionNameFormat, id) + "-valid"
-			subscriberName := fmt.Sprintf(subscriberNameFormat, id) + "-valid"
-
+		It("should mark the Subscription as not ready until NATS is reachable again", func() {
 			ctx := context.Background()
 			natsPort := natsPort + id
```

Comment:
I see what you mean by chaos test now :-D
Moving it to another test suite might a good idea (as you proposed) because currently it shadows the `var natsServer *natsserver.Server` and nats server is started twice

Reply:
Exactly. This test case really feels completely off here.